### PR TITLE
Fix114 ca ExtendedKeyUsage

### LINKED ISSUE #114: truststore certificate verification must specify ExtKeyUsage
by default a x509.Certificate.Verify verifies if a chain exists where each certificate has ExtKeyUsageServerAuth. For client certificates this won't work.

- [x] the verify method must accept a ExtKeyUsage
- [x] CA's must have ExtKeyUsageXXX added.
- [x] CA registration check for ExtKeyUsage
- [x] rfc008 extension

### PR DESCRIPTION
**Pull request by woutslakhorst**
Fixes #114

### CODE REVIEW

File: pkg/x509.go [hunk 1]
```diff
@@ -108,15 +108,31 @@ func (client *Crypto) StoreVendorCACertificate(certificate *x509.Certificate) er
...
+	expectedEKU := []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth}
+	found := 0
+	for _, exp := range expectedEKU {
+		for _, eku := range certificate.ExtKeyUsage {
+			if exp == eku {
+				found++
+			}
+		}
+	}
+	if found != 2 {
```
Comment (reinkrul) [inline pkg/x509.go]: `len(expectedEKU)`

File: pkg/x509.go [hunk 2]
```diff
@@ -108,15 +108,31 @@ func (client *Crypto) StoreVendorCACertificate(certificate *x509.Certificate) er
...
+		return fmt.Errorf("certificate does not define ExtKeyUsage: ClientAuth, ServerAuth")
```
Comment (reinkrul) [inline pkg/x509.go:134]: `fmt.Errorf("certificate does not define ExtKeyUsage: %v", expectedEKU)`

**Review by reinkrul — CHANGES_REQUESTED** [review]

File: pkg/x509.go (hunk 2 above)
Comment (woutslakhorst) [reply pkg/x509.go:134]: x509.ExtKeyUsage are just numbers, so the error will print `[2 1]`

Comment (reinkrul) [reply pkg/x509.go:134]: KeyUsage is a bitmask which Go renders pretty nice (as OR'd components of the bitmask), but hardcoded it is, I guess

**Review by reinkrul — APPROVED** [review]

**Pull request merged**